File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
      int i = 0;
        string ans;

        while (s[i] != '\0') {
            if (isalnum(s[i])) { // Check if alphanumeric
                ans += tolower(s[i]); // Convert to lowercase and append
            }
            i++; // Increment i in every iteration
        }

        string ans1 = ans;
        reverse(ans.begin(), ans.end());
        return ans1 == ans;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        StringBuilder str = new StringBuilder();
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
          if(c>='a' && c<='z' || c >='A' && c <= 'Z' || c>='0' && c<='9'){
           str.append(c);
          }

        }
       
        String str1 = str.toString().toLowerCase();
        str.reverse();
        if(str.toString().toLowerCase().equals(str1)) return true;
        return false;
    }
}